_posts/2016-08-09-the-tech-behind-live-auction-integration.markdown

```diff
@@ -0,0 +1,69 @@
+---
+layout: post
+title: "The Tech Behind Live Auction Integration"
+date: 2016-08-09 11:30
+comments: true
+author: alan
+categories: [engineering, architecture, auctions, javascript, react, redux, scala, akka, swift, launch, lifecycle]
+---
+
+In late June, the [Artsy auctions](https://www.artsy.net/auctions/) team launched our Live Auction Integration (LAI) product. It allows people to participate online in live sales held at auction houses [that partner with Artsy](https://www.artsy.net/auction-partnerships). It was a big project, begun in December, involving both brand new software and extensive integration work with the rest of our systems. This is the first in what will be a series of blog posts discussing the engineering work we did to get a complex product from inception to launch in such a brief time window, with a go-live deadline set in stone weeks in advance. In this, I’ll dig into what we shipped on a high level, as well as some of the overarching technical decisions.
+
+<!-- more -->
+
+LAI raised challenges that were novel to our engineering team. The product is a real-time experience from the perspectives of the bidder and the auction house. Producing that experience requires a complex human + computer system. There are two main flows of information: from auction house to bidder, and vice versa. These are mediated by our systems and staff as follows:
+
+- As bids occur in the auction house sale room, an Artsy operator working on-site inputs that activity into a web interface so that online participants can keep track of what’s happening.
+- As online participants place bids, our system records those as “prospective bids”, and an Artsy bidding clerk on site at the auction house bids on their behalf in the auction house. As those bids are recognized, they are reflected back to all participants, through the prior flow.
+
+To make this easier to visualize:
+
+![Artsy Live Auction Integration Flow](/images/2016-08-09-the-tech-behind-live-auction-integration/Artsy Live Auction Integration flow.png)
```

Comment:
I tend to think the project "code" names unnecessarily distract from this content. It might be more clear if we used short, descriptive labels (like the ones here) but used them exclusively and consistently.

Or, I dunno, maybe it adds useful color.